torch/_decomp/decompositions.py

```diff
@@ -0,0 +1,795 @@
+
+
+@register_decomposition(aten.select_backward)
+def select_backward(grad_output: Tensor, input_sizes: List[int], dim: int, index: int):
+    grad_input = grad_output.new_zeros(input_sizes)
+    return torch.select_scatter(grad_input, grad_output, dim, index)
+
+
+@register_decomposition(aten.diagonal_backward)
+def diagonal_backward(
+    grad_output: Tensor, input_sizes: List[int], offset: int, dim1: int, dim2: int
+):
+    grad_input = grad_output.new_zeros(input_sizes)
+    return torch.diagonal_scatter(grad_input, grad_output, offset, dim1, dim2)
+
+
+@register_decomposition(aten._softmax_backward_data)
+@cast_for_opmath
+def _softmax_backward_data(
+    grad_output: Tensor, output: Tensor, dim: int, input_dtype: int
+):
+    new_grad = grad_output * output
+    return new_grad - output * torch.sum(new_grad, dim=dim, keepdim=True)
+
+
+@register_decomposition(aten._log_softmax_backward_data)
+@cast_for_opmath
+def _log_softmax_backward_data(
+    grad_output: Tensor, output: Tensor, dim: int, input_dtype: int
+):
+    grad_input = grad_output - torch.exp(output) * torch.sum(
+        grad_output, dim=dim, keepdim=True
+    )
+    return grad_input
+
+
+# TODO: the type annotations on arguments are not quite right
+
+
+@register_decomposition(aten.im2col_backward)
+def im2col_backward(
+    grad_output: Tensor,
+    input_size: List[int],
+    kernel_size: List[int],
+    dilation: List[int],
+    padding: List[int],
+    stride: List[int],
+) -> Tensor:
+    return F.fold(grad_output, input_size, kernel_size, dilation, padding, stride)  # type: ignore[arg-type]
+
+
+@register_decomposition(aten.col2im_backward)
+def col2im_backward(
+    grad_output: Tensor,
+    kernel_size: List[int],
+    dilation: List[int],
+    padding: List[int],
+    stride: List[int],
+) -> Tensor:
+    return F.unfold(grad_output, kernel_size, dilation, padding, stride)  # type: ignore[arg-type]
+
+
+@register_decomposition(aten.native_dropout_backward)
+@cast_for_opmath
+def native_dropout_backward(grad_output: Tensor, mask: Tensor, scale: float):
+    return grad_output * (mask.type_as(grad_output) * scale)
+
+
+@register_decomposition(aten.logit_backward)
+@cast_for_opmath
+def logit_backward(
+    grad_output: Tensor, self: Tensor, eps: Optional[float] = None
+) -> Tensor:
+    if eps is not None:
+        lo = eps
+        hi = 1.0 - lo
+        return torch.where(
+            torch.logical_and(self >= lo, self <= hi),
+            grad_output / (self * (1.0 - self)),
+            self.new_zeros(()),
+        )
+    else:
+        return torch.where(
+            torch.logical_and(self >= 0.0, self <= 1.0),
+            grad_output / (self * (1.0 - self)),
+            self.new_full((), float("nan")),
+        )
+
+
+@register_decomposition(aten.native_dropout)
+@cast_for_opmath
+def native_dropout_decomposition(input, p, generator=None):
+    bool_mask = torch.rand_like(input) < p
+    res = bool_mask * input * float(1.0 / p)
```

Comment:
I think these two lines should use (1-p)? cc @ezyang @zou3519 @Chillee @anjali411